common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -668,6 +681,9 @@ private void rethrowIfFailed() {
             return;
         }
 
+        if (!(cause instanceof CancellationException) && ThrowableUtil.getSuppressed(cause).length == 0) {
+            ThrowableUtil.addSuppressed(cause, new RuntimeException("Rethrowing promise failure cause"));
```

Comment:
The `CompletionException` type is only available in Java 8, so I made this a `RuntimeException`.